alica_engine/include/engine/BasicBehaviour.h

```diff
@@ -58,7 +58,7 @@ class BasicBehaviour : private RunnableObject
     using RunnableObject::stop;
     using RunnableObject::TracingType;
 
-    virtual void run() = 0;
+    virtual void run() {}
```

Comment:
`run` become optional instead of mandatory?

Reply:
Yes. there are behaviours with frequency 0 that do all their work at initialization time

Reply:
Maybe we could make child `BasicInitBehaviour` (or BasicRunOnceBehaviour) from `BasicBehaviour`, with only one function `run() {assert(false);}`. I guess it should be like open-close principle

Reply:
There's no real harm in accidentally setting a different frequency with run method empty

Reply:
I'm more concerned about changing interfaces. Especially for some corner case we are changing whole interface behaviour.

Reply:
and yes, calling empty function is harmless, but it could show
1. misconfiguration
2. problems in scheduler

Reply:
It's not a breaking change and it's inconsistent as is so I think the change is appropriate